AstaZero/src/CalculateReferenceLine.cpp

```diff
@@ -22,19 +20,17 @@ CalculateReferenceLine::CalculateReferenceLine(const char* openDriveFileName, do
 				x += reflineOffsetXY[0];
 				y += reflineOffsetXY[1];
 
-				// snprintf(strbuf, sizeof(strbuf), "%lf, %lf, %lf, %lf\n", x, y, 0.0, hdg);
-				// file << strbuf;
-				tmpXYZ.x = x;
-				tmpXYZ.y = y;
-				tmpXYZ.z = 0.0;
-				roadRefLine.point.push_back(tmpXYZ);
-				roadRefLine.hdg.push_back(hdg);
+				tmpXYZHdg.x = x;
+				tmpXYZHdg.y = y;
+				tmpXYZHdg.z = 0.0;
+				tmpXYZHdg.heading = hdg;
+				roadRefLine.push_back(std::make_shared<XYZHdg>(tmpXYZHdg));
```

Comment:
is this really working? Arent you pushing back a pointer to the same object all the time?

Reply:
nop this works to my understanding of how make_shared construct objects, but it might be stupid to do it like this.